### PR TITLE
ReadMe.rst: Add initial file

### PR DESCRIPTION
**Pull request by makubacki**
## Description

Adds the initial readme file for the repo.

It currently provides some background on the project, the
TianoCore UEFI variable driver, and some brief contribution
and code of conduct information.

## How This Was Tested

Verified restructured text output in GitHub preview

## Integration Instructions

N/A - Read the file to gain a better understanding of the repo

Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>

### CODE REVIEW

File: ReadMe.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+Contributing
+============
+
+We welcome everyone to file feature requests, bugs, participate in code reviews, submit code, update documentation,
+and help us build the best variable driver possible.
+
+At this time, we are very early in the work so we're particularly interested in ideas around the future of UEFI
+variables (including breaking UEFI specification compatibility) and suggestions to help shape the overall design.
+
+Please open GitHub issues directly in this repo.
```
Comment (makubacki) [inline ReadMe.rst:23]: @spbrogan, I know most issues are collected in the mu repo. What do you think about trying to use milestones and track feature specific work directly in this repo?

**Review by os-d — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

File: ReadMe.rst (hunk 1 above)
Comment (spbrogan) [reply ReadMe.rst:23]: sounds good.

File: ReadMe.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+Please follow the steps in the Project Mu docs to build for CI and local testing.
+`More Details <https://microsoft.github.io/mu/CodeDevelopment/compile/>`_
+
+Copyright & License
+===================
+
+Some files in this repository have their own copyright. Otherwise, the following copyright applies.
+
+| Copyright (C) Microsoft Corporation
+| SPDX-License-Identifier: BSD-2-Clause-Patent
```
Comment (spbrogan) [inline ReadMe.rst:87]: in my view using github you didn't get a line break between these two lines and it doesn't look good.

Comment (makubacki) [reply ReadMe.rst:87]: I agree the "rich diff" doesn't look great. Does it look better rendered here? https://github.com/makubacki/mu_feature_uefi_variable/tree/add_readme_rst

It looks fine to me outside the diff view.

**Review by kuqin12 — APPROVED** [review]

File: ReadMe.rst (hunk 2 above)
Comment (makubacki) [reply ReadMe.rst:87]: I'm going to go ahead and merge so we have a readme file. I'm happy to make any adjustments, just let me know.

**Pull request merged**